torch/autograd/_functions/linalg.py

```diff
@@ -41,4 +41,12 @@ def forward(self, input):
     def backward(self, grad_output):
         return grad_output.triu(self.diagonal_idx)
 
-# TODO: trace
+
+class Trace(Function):
+
+    def forward(self, input):
+        self.size = input.size()
+        return input.new((input.trace(),))
+
+    def backward(self, grad_output):
+        return grad_output[0] * (torch.eye(*self.size)).type_as(grad_output)
```

Comment:
I think it's better to expand the `grad_output` and call `torch.diag` on it. This implementation will do a large CPU -> GPU transfer, just to create the identity matrix.